server/src/main/java/org/candlepin/model/CertificateSerialCurator.java

```diff
@@ -152,4 +153,26 @@ private Criterion getRevokedCriteria() {
     public Collection<CertificateSerial> saveOrUpdateAll(Map<String, CertificateSerial> serialMap) {
         return this.saveOrUpdateAll(serialMap.values(), false, false);
     }
+
+    @SuppressWarnings("unchecked")
+    public List<Long> listEntitlementSerialIds(Consumer c) {
+        List<Long> resultList = null;
+        String hql = "SELECT s.id" +
+            "    FROM EntitlementCertificate ec" +
+            "     JOIN ec.entitlement e" +
+            "     JOIN e.consumer c" +
+            "     JOIN ec.serial s" +
+            "     JOIN e.pool p" +
+            "    WHERE" +
+            "       c.id=:consumerId" +
+            "    AND" +
+            "       p.endDate >= :nowDate";
```

Comment:
Might want to think about timezones here.  `new Date()` will be in the server's current timezone, I believe.  Not sure what endDate will be stored in (UTC?).  Might be worth some unit tests if you haven't already.